code/game/objects/structures/aliens.dm

```diff
@@ -82,23 +97,146 @@
+	var/is_operating = FALSE
+	var/close_delay = 1.5 SECONDS
+	smoothing_flags = null
+	var/open_sound = 'sound/machines/alien_airlock.ogg'
+	var/close_sound = 'sound/machines/alien_airlock.ogg'
+
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WALLS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WALLS)
 
-/obj/structure/alien/resin/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
-		return !opacity
+/obj/structure/alien/resin/door/Initialize()
+	. = ..()
+	initial_state = icon_state
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	air_update_turf(1)
+	for(var/obj/structure/alien/weeds/node/W in get_turf(src))
+		qdel(W)
+	if(locate(/obj/structure/alien/weeds) in get_turf(src))
+		return
+	new /obj/structure/alien/weeds(loc)
+
+/obj/structure/alien/resin/door/Destroy()
+	density = FALSE
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	playsound(loc, pick('sound/effects/alien_resin_break2.ogg','sound/effects/alien_resin_break1.ogg'), 100, FALSE)
+	air_update_turf(1)
+	return ..()
+
+/obj/structure/alien/resin/door/Move()
+	var/turf/T = loc
+	. = ..()
+	move_update_air(T)
+
+/obj/structure/alien/resin/door/Crossed(mob/living/L, oldloc)
+	..()
+	if(!state_open)
+		return try_to_operate(L)
+
+/obj/structure/alien/resin/door/attack_ai(mob/user)
+	return
+
+/obj/structure/alien/resin/door/attack_hand(mob/user)
+	return try_to_operate(user)
+
+/obj/structure/alien/resin/door/attack_ghost(mob/user)
+	if(user.can_advanced_admin_interact())
+		operate()
+
+/obj/structure/alien/resin/door/CanAtmosPass(turf/T)
 	return !density
 
+/obj/structure/alien/resin/door/proc/try_to_operate(mob/user)
+	if(!is_operating)
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			if(C.get_int_organ(/obj/item/organ/internal/xenos/hivenode))
+				if(world.time - C.last_bumped <= 60)
+					return
+				if(!C.handcuffed)
+					operate()
+			else
+				to_chat(user, "<span class='noticealien'>Your lack of connection to the hive prevents the resin door from opening</span>")
+/*
+  * This 2nd try_to_operate() is needed so that CALLBACK can close the door without having to either call operate() and get bugged when clicked much or
+  * call try_to_operate(atom/user) and not be able to use it due to not having a mob using it
+*/
+/obj/structure/alien/resin/door/proc/mobless_try_to_operate()
+	if(is_operating)
+		if(state_open)
+			addtimer(CALLBACK(src, PROC_REF(mobless_try_to_operate)), close_delay)
+		return
+	operate()
+
+/obj/structure/alien/resin/door/proc/operate()
+	is_operating = TRUE
+	if(!state_open)
+		playsound(loc, open_sound, 100, 1)
+		flick("[initial_state]opening",src)
+	else
+		var/turf/T = get_turf(src)
+		for(var/mob/living/L in T)
+			is_operating = FALSE
+			if(state_open)
+				addtimer(CALLBACK(src, PROC_REF(mobless_try_to_operate)), close_delay)
+			return
+		playsound(loc, close_sound, 100, 1)
+		flick("[initial_state]closing",src)
+	density = !density
+	opacity = !opacity
+	state_open = !state_open
+	sleep(10)
```

Comment:
You might wanna break the below code into another proc with a `1 SECONDS` call delay, to avoid sleeps in code